server/lib/nl/fulfillment_next.py

```diff
@@ -28,11 +30,75 @@
 from server.lib.nl.fulfillment import ranking_across_vars
 from server.lib.nl.fulfillment import simple
 from server.lib.nl.fulfillment import time_delta
+from server.lib.nl.utterance import QueryType
 from server.lib.nl.utterance import Utterance
 
 # We will ignore SV detections that are below this threshold
 _SV_THRESHOLD = 0.5
 
+QUERY_TYPE_HANDLERS = {
+    QueryType.SIMPLE: simple,
+    QueryType.COMPARISON: comparison,
+    QueryType.CONTAINED_IN: containedin,
+    QueryType.RANKING_ACROSS_VARS: ranking_across_vars,
+    QueryType.RANKING_ACROSS_PLACES: ranking_across_places,
+    QueryType.CORRELATION: correlation,
+    QueryType.TIME_DELTA: time_delta,
+    QueryType.EVENT: event,
+    QueryType.OVERVIEW: overview,
+}
+
+# The supported rankings in order. Later entry is preferred.
+RANKED_QUERY_TYPES = [
+    QueryType.SIMPLE,
+    QueryType.COMPARISON,
+    QueryType.CONTAINED_IN,
+    QueryType.RANKING_ACROSS_VARS,
+    QueryType.RANKING_ACROSS_PLACES,
+    QueryType.CORRELATION,
+    QueryType.TIME_DELTA,
+    QueryType.EVENT,
+    QueryType.OVERVIEW,
+]
+
+# The ClassificationTypes not in this map rely on additional fields of detection.
+DIRECT_CLASSIFICATION_TYPE_TO_QUERY_TYPE = {
+    ClassificationType.SIMPLE:
+        QueryType.SIMPLE,
+    ClassificationType.CONTAINED_IN:
+        QueryType.CONTAINED_IN,
+    ClassificationType.CORRELATION:
+        QueryType.CORRELATION,
+    ClassificationType.COMPARISON:
+        QueryType.COMPARISON,
+    ClassificationType.TIME_DELTA:
+        QueryType.TIME_DELTA,
+    ClassificationType.EVENT:
+        QueryType.EVENT,
+
+    # Unsupported classification-types. Map them to SIMPLE for now.
+    # TODO: Handle this better.
+    ClassificationType.UNKNOWN:
+        QueryType.SIMPLE,
+    ClassificationType.OTHER:
+        QueryType.SIMPLE,
+    ClassificationType.TEMPORAL:
+        QueryType.SIMPLE,
+    ClassificationType.CLUSTERING:
+        QueryType.SIMPLE,
+}
+
+DIRECT_QUERY_TYPE_FALLBACK = {
```

Comment:
add a comment here that some have more complex fallback logic?

Reply:
Added (but this is now in a new file)